File: Packets/ConnectionResponse.vb
Public Class ConnectionResponse
    Inherits Packet
    Public Overrides Property OPCode As Byte = &H1
    Public Property Success As Boolean
    Public Property ID As UInteger
    Public Property RejectionReason As String
    Sub New(ByVal _Success As Boolean, ByVal _ID As UInteger, ByVal _RejectionReason As String)
        Success = _Success
        ID = _ID
        RejectionReason = _RejectionReason
    End Sub
    Public Overrides Function GetByteArray() As Byte()
        Dim p As New PacketBuilder
        p.Write(Success)
        If Success Then
            p.Write(VLQ.ToVLQ(ID))
            p.Write(CByte(0))
        Else
            p.Write(VLQ.ToVLQ(0))
            p.Write(RejectionReason)
        End If
        Payload = p.GetBytes
        Return Package(False, True)
    End Function
End Class

File: Packets/ConnectionResponse.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class ConnectionResponse : Packet
{
	public override byte OPCode { get; set; }
	public bool Success { get; set; }
	public int ID { get; set; }
	public string RejectionReason { get; set; }
	public ConnectionResponse(bool _Success, int _ID, string _RejectionReason)
	{
		Success = _Success;
		ID = _ID;
		RejectionReason = _RejectionReason;
	}
	public override byte[] GetByteArray()
	{
		PacketBuilder p = new PacketBuilder();
		p.Write(Success);
		if (Success) {
			p.Write(VLQ.ToVLQ(ID));
			p.Write(Convert.ToByte(0));
		} else {
			p.Write(Convert.ToByte(0));
			p.Write(RejectionReason);
		}
		Payload = p.GetBytes();
		return Package(false, true);
	}
}
